compound_disease/compound_treats_disease/transfer_learning_experiment/transfer_learning.py

```diff
@@ -0,0 +1,270 @@
+#Path subject to change for different os
+database_str = "postgresql+psycopg2://{}:{}@/{}?host=/var/run/postgresql".format(username, password, dbname)
+os.environ['SNORKELDB'] = database_str
+
+from snorkel import SnorkelSession
+session = SnorkelSession()
+
+
+# In[ ]:
+
+
+from snorkel.annotations import LabelAnnotator
+from snorkel.models import candidate_subclass
+
+from metal.label_model import LabelModel
+
+from compound_disease_lf import CD_LFS
+from disease_gene_lf import DG_LFS
+from compound_gene_lf import CG_LFS
+from utils.notebook_utils.dataframe_helper import load_candidate_dataframes
+from utils.notebook_utils.label_matrix_helper import (
+    get_columns, label_candidates, get_auc_significant_stats
+)
+from utils.notebook_utils.plot_helper import plot_label_matrix_heatmap, plot_roc_curve, plot_generative_model_weights
+
+
+# In[ ]:
+
+
+CompoundDisease = candidate_subclass('CompoundDisease', ['Compound', 'Disease'])
+
+
+# In[ ]:
+
+
+quick_load = True
+
+
+# ## Load the Label Matrix
+
+# Before we begin training the generative model, we need to load label matricies for each training/testing set. This process involves extracting annotations from the postgress database shared on this local machine. 
+
+# In[ ]:
+
+
+spreadsheet_names = {
+    'train': 'data/sentence_labels_train.xlsx',
+    'dev': 'data/sentence_labels_train_dev.xlsx',
+    'test': 'data/sentence_labels_dev.xlsx'
+}
+
+
+# In[ ]:
+
+
+candidate_dfs = {
+    key:load_candidate_dataframes(spreadsheet_names[key])
+    for key in spreadsheet_names
+}
+
+for key in candidate_dfs:
+    print("Size of {} set: {}".format(key, candidate_dfs[key].shape[0]))
+
+
+# In[ ]:
+
+
+label_functions = (
+    list(CD_LFS['CtD_DB'].values()) + 
+    list(CD_LFS['CtD_TEXT'].values()) + 
+    list(CD_LFS['CD_BICLUSTER'].values()) + 
+    list(CG_LFS["CbG_TEXT"].values())+
+    list(DG_LFS["DaG_TEXT"].values())
+) 
+
+if quick_load:
+    label_matricies = pickle.load(open("data/label_matricies.pkl", "rb"))
+
+else:
+    label_matricies = {
+        key:label_candidates(
+            session,
+            candidates_dfs[key]['candidate_id'],
+            label_functions,
+            num_threads=10,
+            batch_size=candidates_dfs[key]['candidate_id'].shape[0]
+        )
+        for key in candidate_dfs
+    }
+    
+    pickle.dump(label_matricies, open("data/label_matricies.pkl", "wb"))
+
+
+# In[ ]:
+
+
+lf_names = [
+    label_matricies['test'].get_key(session, index).name 
+    for index in range(label_matricies['test'].shape[1])
+]
+
+
+# ## Observe Label Function Properties
+
+# Before training the generative model, it is a good idea to observe label functions and their output. In the cells below we take a look at generated heatmaps and observe the underlying structure.
+
+# In[ ]:
+
+
+plot_label_matrix_heatmap(label_matricies['train'].T, 
+                          yaxis_tick_labels=lf_names, 
+                          figsize=(10,13))
+
+
+# # Train Generative Model
+
```

Comment:
What are these different models?  I think I might be misunderstanding how you're using this term.  Looks like these are different hyperparams of your model that you're using